lib/mongoid/contextual/memory.rb

```diff
@@ -240,7 +240,7 @@ def limit(value)
       #
       # @param [ String | Symbol ] *fields Field(s) to pluck.
       #
-      # @return [ Array ] The array of plucked values.
+      # @return [ Array<Object> | Array<Array<Object>> ] The plucked values.
```

Comment:
I don't agree with this type annotation... Could you explain why you need the double nested array? Because theoretically we can have n-nested array return values from this method. I think were better off keeping it as just an array.